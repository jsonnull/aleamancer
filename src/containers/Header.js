// @flow
import React from 'react'
import { connect } from 'react-redux'
import { push } from 'react-router-redux'
import Header from 'components/Header'
import { SHOW_SETTINGS } from 'actions/types'
import type { State } from 'store'

type StateProps = {
  username: string
}
const mapStateToProps = (state: State): StateProps => ({
  username: state.user.profile.displayName
})

type DispatchProps = {
  goHome: Function,
  showSettings: Function
}
const mapDispatchToProps = (dispatch: Function): DispatchProps => ({
  goHome: () => dispatch(push('/')),
  showSettings: () => dispatch({ type: SHOW_SETTINGS })
})

export default connect(mapStateToProps, mapDispatchToProps)(Header)
