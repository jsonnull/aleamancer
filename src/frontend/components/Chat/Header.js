// @flow
import React from 'react'
import styled from 'styled-components'
import Tooltip from '../Tooltip'
import { fontSize } from 'frontend/styles/common'

const Container = styled.div`
  background-color: ${props =>
    props.isPinned ? props.theme.background : props.theme.backgroundInverted};
  display: flex;
  flex-direction: row;
`

const Toggle = styled.div`
  display: block;
  font-size: ${fontSize.small};
  color: ${props =>
    props.isPinned ? props.theme.color : props.theme.colorInverted};
  margin-right: 1rem;
  padding-left: 1rem;
  cursor: pointer;

  i {
    transform: ${props => (props.isPinned ? 'none' : 'rotate(45deg)')};
  }
`

const Outer = styled.div`
  margin-left: auto;
`

type Props = {
  isPinned: boolean,
  setChatPinned: Function
}

const Header = (props: Props) => {
  const { isPinned, setChatPinned } = props

  let toggleChat = isPinned ? (
    <i className="fa fa-chevron-left" />
  ) : (
    <i className="fa fa-thumb-tack" />
  )

  const tooltip = isPinned ? 'Unpin Chat' : 'Pin Chat'

  return (
    <Container isPinned={isPinned}>
      <Outer>
        <Tooltip placement="left" content={tooltip}>
          <Toggle onClick={() => setChatPinned(!isPinned)} isPinned={isPinned}>
            {toggleChat}
          </Toggle>
        </Tooltip>
      </Outer>
    </Container>
  )
}

export default Header
