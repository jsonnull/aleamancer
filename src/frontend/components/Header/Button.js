// @flow
import styled from 'styled-components'
import { fontSize } from 'frontend/styles/common'

const Button = styled.div`
  height: 3.25rem;
  width: ${props => (props.square ? '3.25rem' : 'auto')};
  line-height: 3.25rem;
  padding: 0 1rem;
  text-align: center;
  font-size: ${fontSize.normal};
  background: ${props => props.theme.backgroundInverted};
  cursor: pointer;
  border-radius: 5px;
  color: ${props => props.theme.colorInverted};

  &:hover {
    background: ${props => props.theme.backgroundInvertedSecondary};
  }
`

export default Button
