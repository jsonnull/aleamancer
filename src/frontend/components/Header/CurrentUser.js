// @flow
import React from 'react'
import { fontSize } from 'frontend/styles/common'
import Button from './Button'

const SmallButton = Button.extend`
  font-size: ${fontSize.small};
`

type Props = {
  username: string
}

const CurrentUser = (props: Props) => {
  return <SmallButton>{props.username}</SmallButton>
}

export default CurrentUser
