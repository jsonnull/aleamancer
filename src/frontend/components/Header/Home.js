// @flow
import React from 'react'
import { Link } from 'react-router-dom'
import Tooltip from '../Tooltip'
import Button from './Button'

const Home = () => (
  <Tooltip placement="right" content="Home">
    <Link to="/sessions">
      <Button square>
        <i className="fa fa-home" />
      </Button>
    </Link>
  </Tooltip>
)

export default Home
