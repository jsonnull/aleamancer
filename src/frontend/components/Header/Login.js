// @flow
import React from 'react'
import { Link } from 'react-router-dom'
import Button from './Button'

const Login = () => (
  <Link to="/login" style={{ textDecoration: 'none' }}>
    <Button>Login</Button>
  </Link>
)

export default Login
