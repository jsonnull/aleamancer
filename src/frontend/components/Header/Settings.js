// @flow
import React from 'react'
import Button from './Button'

type Props = {
  showSettings: Function
}

const Settings = (props: Props) => {
  const { showSettings } = props

  return (
    <Button square onClick={showSettings}>
      <i className="fa fa-cog" />
    </Button>
  )
}

export default Settings
