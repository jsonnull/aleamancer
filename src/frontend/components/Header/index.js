// @flow
import * as React from 'react'
import styled from 'styled-components'
import { CONSTS } from 'frontend/styles/common'
import Home from './Home'
import Logo from './Logo'
import CurrentUser from './CurrentUser'
import Settings from './Settings'
import Login from './Login'

const Container = styled.div`
  display: flex;
  flex-direction: row;
  height: ${CONSTS.headerHeight};
  line-height: ${CONSTS.headerHeight};
  color: white;
  background-color: ${props => props.theme.backgroundInverted};
`

const Column = styled.div`
  display: flex;
  flex-direction: row;
  flex: 1;
  padding: 0.5rem;
`

const Right = Column.extend`
  justify-content: flex-end;
`

type Props = {
  username: string | null,
  showSettings: Function
}

const Header = (props: Props) => {
  const { username, showSettings } = props

  return (
    <Container>
      <Column>
        <Home />
      </Column>
      <Column>
        <Logo />
      </Column>
      <Right>
        {username === null ? (
          <Login />
        ) : (
          <React.Fragment>
            <CurrentUser username={username} />
            <Settings showSettings={showSettings} />
          </React.Fragment>
        )}
      </Right>
    </Container>
  )
}

export default Header
