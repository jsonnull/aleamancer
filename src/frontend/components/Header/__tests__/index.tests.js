// @flow
import React from 'react'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router'
import Header from '../index.js'

describe('Header component', () => {
  it('renders correctly with no user', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Header username={null} showSettings={() => {}} />
        </MemoryRouter>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly when logged in', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Header username="test" showSettings={() => {}} />
        </MemoryRouter>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
