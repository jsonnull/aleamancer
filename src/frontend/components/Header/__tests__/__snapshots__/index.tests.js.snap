// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders correctly when logged in 1`] = `
.c2 {
  height: 3.25rem;
  width: 3.25rem;
  line-height: 3.25rem;
  padding: 0 1rem;
  text-align: center;
  font-size: 1.5rem;
  cursor: pointer;
  border-radius: 5px;
}

.c4 {
  height: 3.25rem;
  width: auto;
  line-height: 3.25rem;
  padding: 0 1rem;
  text-align: center;
  font-size: 1.5rem;
  cursor: pointer;
  border-radius: 5px;
}

.c5 {
  position: relative;
  top: 10px;
  width: 82px;
  height: 12px;
  background-image: url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAFIAAAAMCAQAAABboSKXAAAAAnNCSVQICFXsRgQAAAAJcEhZcwAADsQAAA7EAZUrDhsAAAAZdEVYdFNvZnR3YXJlAHd3dy5pbmtzY2FwZS5vcmeb7jwaAAADBUlEQVQ4y83US4iXZRTH8Z+NN7QIRU3EcUJEK3NjNYKMaTRaEFJZSSVechYqShpFSRJq0kYtJbAyLNsUJTnhpZuQWaG5EEscywtvQhcYUYbUvKTOfFrM69/BiLae1XPO+b3P+Z7Ded6kYl5SWFXxulpjpzW5lkwXzTilRyVyCIuuLchJ2u3JSuQrzLm2ILfjIL6uRDZjWnkeo8E4nZLEAI9rMLbM3KveIDWmGpeoMsFUfcvcaDNNMaj8ql69KnUa3F6p0Vm9Bo+pThLDzPCg6xOd1KtXrbvJ7bkkavGnIf7GiDK2qX2uevoSZ7EhSXxXzvyDJNGElf4CL1gPjumXGF7qzpmQqMNFj2rFJZOSxChHS83MxFJtzuGwPonDmGU36i9DfoxXEx9i7VWQK7AksRhj3WCpceZoxc2JrdjhaTvQYrHnnMbsxETT1NmCLxIDccFqCzRhe+Imx13ylGGWecD92K6rOm2WJbZhv33+MLQdZ6hWF9UkxuGMXlcgdXIC9xhsPOaXDXRzFHVle9MTt+F0kljX3lSpnIgDiT44m5Q1DiYW4pOKbgMWGWywZpvLZXs+0UVVu2AtLigUfgHPdoDs7bKd0mJ+opcljoExHSD7VyCXX4Y0ykaX/gU5EocSH+G1CuTeynK0aCwhn7jyZPo7h7322GOPwyhcV4HsrhW3VNSjnfCWekf+D9IqbSaa+5+Q76Oxcu83mNvhIV8FuRLfV7zezuORxBZMSRzBOzonqg32LWaUnd+daCz9AThTue9lw3FGlVr8nOiH80niThxJLMAFtWXdN7Ffr0RPd5W7fvl3aKCzmNWhh0b8oMouzEvMAb/70QUDfYpfvaIZD5cTeCYxAm26JdbjdTVasdU6HEvcCnok7sPxxI0O4bxN3jXeCOdw0h4nTU/s6jBZbysURnaAXKBQmK1JYUWSmG63U1q8lxhim2Yv2qqwUJV9CqsTkxUKdyQ+V9iYaPCTA2YqFGo8pFCoTcxTKFQn+nrDb87bqXNilM80O2OXfrpoUljeTvQPR26bCLlSuF4AAAAASUVORK5CYII=);
  background-size: 82px 12px;
}

.c3 {
  margin: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c7 {
  height: 3.25rem;
  width: auto;
  line-height: 3.25rem;
  padding: 0 1rem;
  text-align: center;
  font-size: 1.5rem;
  cursor: pointer;
  border-radius: 5px;
  font-size: 1.3rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: 4.25rem;
  line-height: 4.25rem;
  color: white;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 0.5rem;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 0.5rem;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

@media only screen and (-webkit-min-device-pixel-ratio:1.25), @media only screen and (-moz-min-device-pixel-ratio:1.25), @media only screen and (-o-min-device-pixel-ratio:5/4), @media only screen and (min-device-pixel-ratio:1.25) {
  .c5 {
    background-image: url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAKQAAAAYCAQAAADetJ7EAAAAAnNCSVQICFXsRgQAAAAJcEhZcwAAHYcAAB2HAY/l8WUAAAAZdEVYdFNvZnR3YXJlAHd3dy5pbmtzY2FwZS5vcmeb7jwaAAAGq0lEQVRYw+2ZZ3BVVRDHN4mJAQkDDMUCg6JgQccWUSPFAuNIG7GBCCoIUVGkyKCODXWQgDSNMzSRNoCKAyhDQEQZiSWQgKAhUgxSRJTBEDAmBEh+fnh79932HiB8zHlf7tn/nr33/N4pe+4VCS3kUEAB+dQPVdNoy7N8QAEjpKbELtyIU/qGqEkcMv2BGlrxQI4zUItD9UWmX1NDKzbGVA4YqEqahHgMM/28Gl6xQfbHXV4M8XhMtRM1tOKB3OQBuZPEgEd31UpraMXGeKdCOk61XnUO+Nyhyr4aXrFBfqGQFrJGr3ICPrerss1nv5b3WE0huYzlYo9yM2+zgg1sYS3vcKVL6cA0/XUSoQ3z2MT3vEsz1W9lPpvZyAxaeyLWYyALyaWIAj6iD8kubZJGnCAiwsMsp5B1TKR5SH+bMoLFbGAb+SxhuEupz/MsZRP5zHMPJkbYE9cToQ6jWM08f9jrbEq3oadeVdEyBsgil602CzxLQgX3m5bM3x7tOE+7EDulPz04brUyOojQkxNmOUp31/1ewlvW08C0r9W2w7WeA5SS4cuHsznmiZJnWh/KPMp8569istlq04QdYUNKDMa3IqTwp9Ymnwwk57LOAO6xjl+taif8pYo2Bvmo2gZQ4vHZyvkc9lhKSLM75gdiLjJttIOF9lR6fHaTYl5N2ByIsVy14VqvZpdFeEu1ntbTc8jT6y+9gC6xEXGf54FKqXMSkGPVMpNzRWjJTgA+UXUKh5jJMF7kO3vk2dZ2r1p+5RhF/OEZZaXkstVl6aVtmlPNt0xiKNmUW6cvVPUZtfzGQcopotQV4R47VjjPspdM0unKp8AcnZcReDtoJUIqMwEoj4x52tmcGWJRZ3pBZqu5mCR9XGdiDYoHkhbqt59aaumrt07WaVJX7Ql8pm1/DmQJs2gkQoJr0s6L5Kk8ZBvfm9om3RnRIrSlStWuauml9b94kCQRknjBn87xhPk0s0hdeF1ExHYHXUqoqzOjt4gIV6l6jMNsZBxT2c0oN55G/KsuQ8y2TC1bSIgD8jUHhXk0VEvrwOLeVZXfzeIsCt20nsh2tYw0n6/CFxlVf1K1j9Z7aP07l48DJ0vr67U+zBMpQYTmdhixDYzl0bZcasvT4AgVUmjqDvKGTeS0QLehYxyQK2xid9RfJ10k2vq6nGKrz75YIEVYrJYXzPJubJBczi+qPhoHpBNhrIgIl0U31UA8Zzzvsd501B5O94AsC097zuOgOhymwH4/2g2XxgG5iVglwzPiX2afKfFAzg6AHB0GkgTuJsemfXyQ4z0gO1ubwBHY/upgmXoqIJ8jfjkRzQwDIPd7PMspoYRiitlCK/VoTDYVHq8zBEkC97PF94ynDjLT2tQNkBjr63cJJeylmB2RLDMuSJLZxcnKuJggnX33lRhJfj+OACvJIotZZwMkF7AWKCWLLLI0RzgdkE9Zr64MPO1oOxwnh/YmLsg+FniO5e3Oz1nK/6b2Sab2x6E3HqzqdSIi3HDmIGmguepHqq48bZD3Wn8HxZza1TQ6TZAkUqhibkjDaDr9ZAyQH2r9XyeTc7WuwxH3anRWQI7R2qT/DbKZraxbSY3xJsGVvZwqyEesaW8Jw7zbO9i520mhfbk+rI8eJ7mHZSRyk2ntPSAPBvLI6JFyoVpeNcsEtUwREWG11lb4QA707bv5rl6871scvrfnWhg97YiIUMu+AJTT06zNmR7JXKJ5pB/UOZa3HXISap/H23bTfp6FoMQi7HS9Cs5hEtMoBOaKcL0pm8kg1cBWRZJ+ETvLDLT7rfaOOBHmuhcPS7eqGUljl/9LvsWk2NWHj9U218bdCddhdDAZpNONMaS4xj/Aj8xgIiuppDxChw6mpXkxPRs9ScTYLNLN4zfSRBhpMFLtHU1lyPbUW4Qk24qczjvlQs0sj/m3KorUsiCA9pvQDMOJma3eztT/J/oulVy1rTLLkJAnHqXfCPJDtJXa7mGzuF/mcJWdZ+wfDckxo92f6xoRdnIV4TYfMKiioYgId7nu8AO97foJ34q0RiNdZHfbHwFBqkU4Sh0RUuxlH1QwhFV6vV0j5Jl6i73nqbQTsq2JZHpO4XDAGWPUs6NstAxVbZp/z3A2imj6/VAMkImsd3ndRq5dD/d868nkc/ZSQSlFzKKLKS2YSB7FTCOFK6zteE1EnHpepIt0dd2rpa6rUUuGvnJ4nGX8wmbaiJBt6sUi1Gad1QfYeTwaId31zA3IJIdiKihju/ebKe2YTiGHOcouvmAw9dS+xCKNi3r/Bx8J5tkORG3bAAAAAElFTkSuQmCC);
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <div>
        <a
          href="/sessions"
          onClick={[Function]}
        >
          <div
            className="c2"
          >
            <i
              className="fa fa-home"
            />
          </div>
        </a>
      </div>
    </div>
  </div>
  <div
    className="c1"
  >
    <div
      className="c3"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <div
          className="c4"
        >
          <div
            className="c5"
          />
        </div>
      </a>
    </div>
  </div>
  <div
    className="c6"
  >
    <div
      className="c7"
    >
      test
    </div>
    <div
      className="c2"
      onClick={[Function]}
    >
      <i
        className="fa fa-cog"
      />
    </div>
  </div>
</div>
`;

exports[`Header component renders correctly with no user 1`] = `
.c2 {
  height: 3.25rem;
  width: 3.25rem;
  line-height: 3.25rem;
  padding: 0 1rem;
  text-align: center;
  font-size: 1.5rem;
  cursor: pointer;
  border-radius: 5px;
}

.c4 {
  height: 3.25rem;
  width: auto;
  line-height: 3.25rem;
  padding: 0 1rem;
  text-align: center;
  font-size: 1.5rem;
  cursor: pointer;
  border-radius: 5px;
}

.c5 {
  position: relative;
  top: 10px;
  width: 82px;
  height: 12px;
  background-image: url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAFIAAAAMCAQAAABboSKXAAAAAnNCSVQICFXsRgQAAAAJcEhZcwAADsQAAA7EAZUrDhsAAAAZdEVYdFNvZnR3YXJlAHd3dy5pbmtzY2FwZS5vcmeb7jwaAAADBUlEQVQ4y83US4iXZRTH8Z+NN7QIRU3EcUJEK3NjNYKMaTRaEFJZSSVechYqShpFSRJq0kYtJbAyLNsUJTnhpZuQWaG5EEscywtvQhcYUYbUvKTOfFrM69/BiLae1XPO+b3P+Z7Ded6kYl5SWFXxulpjpzW5lkwXzTilRyVyCIuuLchJ2u3JSuQrzLm2ILfjIL6uRDZjWnkeo8E4nZLEAI9rMLbM3KveIDWmGpeoMsFUfcvcaDNNMaj8ql69KnUa3F6p0Vm9Bo+pThLDzPCg6xOd1KtXrbvJ7bkkavGnIf7GiDK2qX2uevoSZ7EhSXxXzvyDJNGElf4CL1gPjumXGF7qzpmQqMNFj2rFJZOSxChHS83MxFJtzuGwPonDmGU36i9DfoxXEx9i7VWQK7AksRhj3WCpceZoxc2JrdjhaTvQYrHnnMbsxETT1NmCLxIDccFqCzRhe+Imx13ylGGWecD92K6rOm2WJbZhv33+MLQdZ6hWF9UkxuGMXlcgdXIC9xhsPOaXDXRzFHVle9MTt+F0kljX3lSpnIgDiT44m5Q1DiYW4pOKbgMWGWywZpvLZXs+0UVVu2AtLigUfgHPdoDs7bKd0mJ+opcljoExHSD7VyCXX4Y0ykaX/gU5EocSH+G1CuTeynK0aCwhn7jyZPo7h7322GOPwyhcV4HsrhW3VNSjnfCWekf+D9IqbSaa+5+Q76Oxcu83mNvhIV8FuRLfV7zezuORxBZMSRzBOzonqg32LWaUnd+daCz9AThTue9lw3FGlVr8nOiH80niThxJLMAFtWXdN7Ffr0RPd5W7fvl3aKCzmNWhh0b8oMouzEvMAb/70QUDfYpfvaIZD5cTeCYxAm26JdbjdTVasdU6HEvcCnok7sPxxI0O4bxN3jXeCOdw0h4nTU/s6jBZbysURnaAXKBQmK1JYUWSmG63U1q8lxhim2Yv2qqwUJV9CqsTkxUKdyQ+V9iYaPCTA2YqFGo8pFCoTcxTKFQn+nrDb87bqXNilM80O2OXfrpoUljeTvQPR26bCLlSuF4AAAAASUVORK5CYII=);
  background-size: 82px 12px;
}

.c3 {
  margin: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: 4.25rem;
  line-height: 4.25rem;
  color: white;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 0.5rem;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 0.5rem;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

@media only screen and (-webkit-min-device-pixel-ratio:1.25), @media only screen and (-moz-min-device-pixel-ratio:1.25), @media only screen and (-o-min-device-pixel-ratio:5/4), @media only screen and (min-device-pixel-ratio:1.25) {
  .c5 {
    background-image: url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAKQAAAAYCAQAAADetJ7EAAAAAnNCSVQICFXsRgQAAAAJcEhZcwAAHYcAAB2HAY/l8WUAAAAZdEVYdFNvZnR3YXJlAHd3dy5pbmtzY2FwZS5vcmeb7jwaAAAGq0lEQVRYw+2ZZ3BVVRDHN4mJAQkDDMUCg6JgQccWUSPFAuNIG7GBCCoIUVGkyKCODXWQgDSNMzSRNoCKAyhDQEQZiSWQgKAhUgxSRJTBEDAmBEh+fnh79932HiB8zHlf7tn/nr33/N4pe+4VCS3kUEAB+dQPVdNoy7N8QAEjpKbELtyIU/qGqEkcMv2BGlrxQI4zUItD9UWmX1NDKzbGVA4YqEqahHgMM/28Gl6xQfbHXV4M8XhMtRM1tOKB3OQBuZPEgEd31UpraMXGeKdCOk61XnUO+Nyhyr4aXrFBfqGQFrJGr3ICPrerss1nv5b3WE0huYzlYo9yM2+zgg1sYS3vcKVL6cA0/XUSoQ3z2MT3vEsz1W9lPpvZyAxaeyLWYyALyaWIAj6iD8kubZJGnCAiwsMsp5B1TKR5SH+bMoLFbGAb+SxhuEupz/MsZRP5zHMPJkbYE9cToQ6jWM08f9jrbEq3oadeVdEyBsgil602CzxLQgX3m5bM3x7tOE+7EDulPz04brUyOojQkxNmOUp31/1ewlvW08C0r9W2w7WeA5SS4cuHsznmiZJnWh/KPMp8569istlq04QdYUNKDMa3IqTwp9Ymnwwk57LOAO6xjl+taif8pYo2Bvmo2gZQ4vHZyvkc9lhKSLM75gdiLjJttIOF9lR6fHaTYl5N2ByIsVy14VqvZpdFeEu1ntbTc8jT6y+9gC6xEXGf54FKqXMSkGPVMpNzRWjJTgA+UXUKh5jJMF7kO3vk2dZ2r1p+5RhF/OEZZaXkstVl6aVtmlPNt0xiKNmUW6cvVPUZtfzGQcopotQV4R47VjjPspdM0unKp8AcnZcReDtoJUIqMwEoj4x52tmcGWJRZ3pBZqu5mCR9XGdiDYoHkhbqt59aaumrt07WaVJX7Ql8pm1/DmQJs2gkQoJr0s6L5Kk8ZBvfm9om3RnRIrSlStWuauml9b94kCQRknjBn87xhPk0s0hdeF1ExHYHXUqoqzOjt4gIV6l6jMNsZBxT2c0oN55G/KsuQ8y2TC1bSIgD8jUHhXk0VEvrwOLeVZXfzeIsCt20nsh2tYw0n6/CFxlVf1K1j9Z7aP07l48DJ0vr67U+zBMpQYTmdhixDYzl0bZcasvT4AgVUmjqDvKGTeS0QLehYxyQK2xid9RfJ10k2vq6nGKrz75YIEVYrJYXzPJubJBczi+qPhoHpBNhrIgIl0U31UA8Zzzvsd501B5O94AsC097zuOgOhymwH4/2g2XxgG5iVglwzPiX2afKfFAzg6AHB0GkgTuJsemfXyQ4z0gO1ubwBHY/upgmXoqIJ8jfjkRzQwDIPd7PMspoYRiitlCK/VoTDYVHq8zBEkC97PF94ynDjLT2tQNkBjr63cJJeylmB2RLDMuSJLZxcnKuJggnX33lRhJfj+OACvJIotZZwMkF7AWKCWLLLI0RzgdkE9Zr64MPO1oOxwnh/YmLsg+FniO5e3Oz1nK/6b2Sab2x6E3HqzqdSIi3HDmIGmguepHqq48bZD3Wn8HxZza1TQ6TZAkUqhibkjDaDr9ZAyQH2r9XyeTc7WuwxH3anRWQI7R2qT/DbKZraxbSY3xJsGVvZwqyEesaW8Jw7zbO9i520mhfbk+rI8eJ7mHZSRyk2ntPSAPBvLI6JFyoVpeNcsEtUwREWG11lb4QA707bv5rl6871scvrfnWhg97YiIUMu+AJTT06zNmR7JXKJ5pB/UOZa3HXISap/H23bTfp6FoMQi7HS9Cs5hEtMoBOaKcL0pm8kg1cBWRZJ+ETvLDLT7rfaOOBHmuhcPS7eqGUljl/9LvsWk2NWHj9U218bdCddhdDAZpNONMaS4xj/Aj8xgIiuppDxChw6mpXkxPRs9ScTYLNLN4zfSRBhpMFLtHU1lyPbUW4Qk24qczjvlQs0sj/m3KorUsiCA9pvQDMOJma3eztT/J/oulVy1rTLLkJAnHqXfCPJDtJXa7mGzuF/mcJWdZ+wfDckxo92f6xoRdnIV4TYfMKiioYgId7nu8AO97foJ34q0RiNdZHfbHwFBqkU4Sh0RUuxlH1QwhFV6vV0j5Jl6i73nqbQTsq2JZHpO4XDAGWPUs6NstAxVbZp/z3A2imj6/VAMkImsd3ndRq5dD/d868nkc/ZSQSlFzKKLKS2YSB7FTCOFK6zteE1EnHpepIt0dd2rpa6rUUuGvnJ4nGX8wmbaiJBt6sUi1Gad1QfYeTwaId31zA3IJIdiKihju/ebKe2YTiGHOcouvmAw9dS+xCKNi3r/Bx8J5tkORG3bAAAAAElFTkSuQmCC);
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <div>
        <a
          href="/sessions"
          onClick={[Function]}
        >
          <div
            className="c2"
          >
            <i
              className="fa fa-home"
            />
          </div>
        </a>
      </div>
    </div>
  </div>
  <div
    className="c1"
  >
    <div
      className="c3"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <div
          className="c4"
        >
          <div
            className="c5"
          />
        </div>
      </a>
    </div>
  </div>
  <div
    className="c6"
  >
    <a
      href="/login"
      onClick={[Function]}
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      <div
        className="c4"
      >
        Login
      </div>
    </a>
  </div>
</div>
`;
